cpp/src/arrow/compute/kernels/generated/codegen.py

```diff
@@ -78,6 +78,8 @@ def generate(self):
     CastCodeGenerator('Int64', NUMERIC_TYPES + STRING_TYPES),
     CastCodeGenerator('Float', NUMERIC_TYPES + STRING_TYPES),
     CastCodeGenerator('Double', NUMERIC_TYPES + STRING_TYPES),
+    CastCodeGenerator('Decimal128', ['Decimal128', 'Int64'],
```

Comment:
Only `Int64`? I would expect all signed integer types.

Reply:
Added other ints though they are not really needed as there is cast from int64 to them.

Reply:
Added Int32 Int16 Int8